SWLOR.Game.Server/Item/RepairKit.cs

```diff
@@ -31,20 +33,47 @@ public CustomData StartUseItem(NWCreature user, NWItem item, NWObject target, Lo
         public void ApplyEffects(NWCreature user, NWItem item, NWObject target, Location targetLocation, CustomData customData)
         {
             SkillType skillType = GetSkillType(item);
+            NWItem targetitem = (target.Object);
             int tech = item.GetLocalInt("TECH_LEVEL");
             float maxDurabilityReductionPenalty = item.GetLocalFloat("MAX_DURABILITY_REDUCTION_PENALTY");
             int repairAmount = tech * 2;
-
+            int skillrank;
+            int Level = targetitem.RecommendedLevel;
+            int delta = 0;
+            int baseXP = 0;
             if (skillType == SkillType.Armorsmith)
-                repairAmount += item.CraftBonusArmorsmith + (_perk.GetPCPerkLevel(user.Object,PerkType.ArmorRepair) * 2);
+            {
+                skillrank = (_skill.GetPCSkillRank((NWPlayer)user, skillType));
```

Comment:
Our NW object wrappers suck. To be on the safe side, do user.Object instead of a direct cast. It's one of the things I want to improve. I just haven't had time.